src/pyggdrasil/tree_inference/_mcmc.py

```diff
@@ -184,7 +186,9 @@ def _swap_subtrees_proposal(key: random.PRNGKeyArray, tree: Tree) -> tuple[Tree,
         desc_node2 = tr._get_descendants(tree.tree_topology, tree.labels, node2)
         # \Delta q = log q(new|old) - log q(old|new) = log [d(i)+1] - log [d(k)+1]
         # where k is the descendant node of i
-        corr = float(jnp.log(desc_node1 + 1.0) - jnp.log(desc_node2 + 1.0))
+        corr = float(
+            jnp.log(desc_node1.shape[0] + 1.0) - jnp.log(desc_node2.shape[0] + 1.0)
```

Comment:
Here it doesn't matter (as shapes are integers), but perhaps in some of your future projects you'll find this function useful: https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.log1p.html
(It's primary use case is for `log(1+p)` for small `p`).